themes/bootstrap3/templates/content/faq.phtml

```diff
@@ -12,3 +12,12 @@
   Q: Why does the service use cookies?<br/>
   A: Cookies are used to manage sessions, which include selected language, last search, book bag contents, any login information, etc.
 </p>
+<p>
+  Q: What kinds of services have been built on top of VuFind?<br>
```

Comment:
Is this a temporary test or something that's really meant to be included? If we're keeping it, we need the (R) symbol on VuFind (though it looks like a test to me). :-)

Reply:
That's just a test for iframemanager, I'll need to think about it. It's also possible to leave out the iframemanager support, but I think that it can be a relatively popular choice to include some videos or social media content e.g. on the front page.

Reply:
Makes sense. Maybe we just need a comment somewhere to clarify that VuFind doesn't use YouTube for anything by default, but this setting supports local use where applicable.

Reply:
Maybe it's enough to document in the wiki how it works so that no embedded video or such is included in the core package. At least the configuration should be available since it took some effort to list all the cookies YouTube uses (and I'm still not sure I didn't miss any!).

Reply:
I guess a related question (which may not be answerable without the experience gained by time) is, "how fragile is this?" I would imagine that YouTube could change its cookie strategy at any time without notice and break things. Perhaps that's an argument that we should put service-specific settings on a wiki page, and put a link to that wiki page in the config file. Then, if people want to add something, they can find the latest working settings easily, even if they're running a slightly out-of-date VuFind release. And then we could set up a directory of other services like Vimeo, etc., as needed. (Though I still could be persuaded to keep these in if you have strong feelings about it -- still in brainstorming mode at the moment).

Reply:
That is an issue, but apparently unavoidable. I've discussed this very issue with our specialist and he said that we must display all the information in the cookie consent preferences. I suppose that we can only do our best. From what I've gathered, there are quite a few cookie consent dialogs around with less cookies for YouTube, which shows how difficult it is to get right.

Reply:
Also, some of the cookies, or at least their expiration times, are regional or dependent on whether the user is logged in to Google services.

Reply:
Understood! Do you have a preference at this point about whether to lean toward the wiki or the built-in configuration? If you like the idea of the wiki, please feel free to start building out pages whenever it's convenient (and let me know if you need help establishing anything or deciding where to put things).

Reply:
Built-in is a must for us. Otherwise we'd have to somehow embed the wiki page into the consent preferences dialog, which would be ugly and dependent on the wiki being available. Making the user go to a third-party site, no matter how friendly, is a no-no.

Reply:
Regardless, a wiki page is needed for instructions on how to set up additional services and verify current settings.

Reply:
I think I may not have clearly expressed my idea -- I wasn't talking about redirecting end users to our wiki; I was proposing that we have YAML sections pre-built in the wiki that can be copied and pasted into the local YAML, so we can publish configurations for a variety of third-party services users might potentially want to embed in VuFind, without having to include them in a single huge file in the project... and that would allow us to keep them up to date without having to issue new VuFind releases. Then the YAML file in the project could just reference the wiki and suggest pulling in whatever section(s) are needed.

Reply:
...but maybe there are compelling reasons to argue FOR including everything in the project, in which case I suppose we could also include a comment pointing to the current version of the file in GitHub and suggest that people watch it for changes if they're using older releases.

Reply:
Sorry, I didn't get it! That'd make sense. That leaves the question whether it makes sense to include something in the distributed yaml file as an example or leave it out.

Reply:
Maybe if we include an example, it could be a fictitious one (pointing at a myuniversity.edu site, or something) so we don't need to worry about it getting out of date and causing confusion... and as long as we have a comment pointing at the wiki for real settings, hopefully that will allow people to get what they need fairly easily.

Reply:
One potential pitfall with having yaml in the wiki is that we need to make sure not to introduce changes that are incompatible with older versions. I suppose that shouldn't be a huge issue since we try to maintain back-compatibility for configuration in any case.

Reply:
Yes, no solution is entirely foolproof, but as you say, this is just a further reason to be careful about backward-compatibility, which we already try to do. And worst case scenario where something changes dramatically, we can always put a note on the top of the wiki saying "These notes require version X at a minimum" and link to a prior revision for legacy users.

Reply:
I've now created the initial [wiki page](https://vufind.org/wiki/configuration:cookie_consent) with YouTube sample config.

On a second thought I'm not sure how useful it would be to include an example in the default yaml, so maybe it's easier to just leave it out since it also reduces the number of translation strings.

Reply:
Thanks, it looks good. I agree that reducing the amount of extra translation work we need to do is helpful!